### PR TITLE
soundpipe 1.7.0 (new-formula)

### PR DESCRIPTION
**Pull request by gabcoh**
Add a formula for the soundpipe library at
https://github.com/PaulBatchelor/soundpipe/
The library is installed just by make and make install

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/soundpipe.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Soundpipe < Formula
+  desc "Lightweight music DSP library"
+  homepage "https://paulbatchelor.github.io/proj/soundpipe.html"
+  url "https://github.com/PaulBatchelor/soundpipe/archive/v1.7.0.tar.gz"
+  sha256 "2d6f6b155ad93d12f59ae30e2b0f95dceed27e0723147991da6defc6d65eadda"
+  depends_on "libsndfile"
```
Comment (fxcoudert) [inline Formula/soundpipe.rb]: Blank line before `depends_on`

File: Formula/soundpipe.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def install
+    system "make"
+    system "make", "-C", "test"
+    cp_r "test", prefix
+    cp_r "examples", prefix
+    system "make", "install", "PREFIX="+prefix
```
Comment (fxcoudert) [inline Formula/soundpipe.rb]: `PREFIX=#{prefix}`

File: Formula/soundpipe.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  test do
+    # TODO
+    # While this command does test the library, it does so by printing out ok or
+    # not ok for various tests. A more thorough test would check to make sure
+    # that everything is ok before returning success.
+    system "cd #{prefix}/test;#{prefix}/test/run.bin"
```
Comment (fxcoudert) [inline Formula/soundpipe.rb]: Please check the output with `assert_match` and `shell_output`.

File: Formula/soundpipe.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Lightweight music DSP library"
+  homepage "https://paulbatchelor.github.io/proj/soundpipe.html"
+  url "https://github.com/PaulBatchelor/soundpipe/archive/v1.7.0.tar.gz"
+  sha256 "2d6f6b155ad93d12f59ae30e2b0f95dceed27e0723147991da6defc6d65eadda"
+  depends_on "libsndfile"
+
+  def install
+    system "make"
+    system "make", "-C", "test"
+    cp_r "test", prefix
```
Comment (fxcoudert) [inline Formula/soundpipe.rb]: tests and examples should be installed to `pkgshare`.

File: Formula/soundpipe.rb (hunk 3 above)
Comment (fxcoudert) [reply Formula/soundpipe.rb]: Another option would be to compile and link a simple C app, if the project ships a library.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by gabcoh** [comment]
Sorry for the newbie question but how should I make these changes? Just with another commit in my branch or make a new pull request?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @gabcoh for the pull request!

**Pull request closed without merging**